src/app/qgsidentifyresultsdialog.cpp

```diff
@@ -2324,6 +2326,35 @@ void QgsIdentifyResultsDialog::copyFeatureAttributes()
   clipboard->setText( text );
 }
 
+void QgsIdentifyResultsDialog::selectFeatureByAttribute()
+{
+  QTreeWidgetItem *item = lstResults->currentItem();
+  if ( !item ) // should not happen
+  {
+    QgsDebugMsg( QStringLiteral( "Selected item is not feature" ) );
+    return;
+  }
+
+  const QString attribute = item->data( 0, Qt::DisplayRole ).toString();
+  const QString value = item->data( 1, Qt::DisplayRole ).toString();
+
+  QgsVectorLayer *vlayer = vectorLayer( item );
+
+  if ( !vlayer )
+    return;
+
+  if ( value != QString( "NULL" ) )
```

Comment:
You should use QgsExpression::createFieldEqualityExpresssion here instead -- all the logic is already there

Reply:
Ah, thanks @nyalldawson!

I didn't know we have that method. But, as far as I check, this doesn't work with a list data type. It seems the `=` operator doesn't work for a list data type. For example:

This one doesn't work:
```
"String List" = array('z', 'y', 'x')
```

I found a way to compare a list, but I am not sure if this is the correct way:
```
 array_to_string( "String List") = array_to_string( array('z', 'y', 'x') ) 
```

Reply:
Hmm, there's logic in place to test for list equality in expressions -- see https://github.com/qgis/QGIS/blob/e0a961fe52e3c222af19d48abb48f25d171ba744/src/core/expression/qgsexpressionnodeimpl.cpp#L358

So "String List" = array('z', 'y', 'x') should work. Might be time to fire up your debugger and see what's missing here 👍

Reply:
Could also be that the stringlist type needs to be explicitly handled in https://github.com/qgis/QGIS/blob/e0a961fe52e3c222af19d48abb48f25d171ba744/src/core/expression/qgsexpression.cpp#L102

Reply:
@ismailsunni fixed in https://github.com/qgis/QGIS/pull/47558

Reply:
whoa, thanks a lot @nyalldawson